changelog/13899.changes.rst

```diff
@@ -0,0 +1,4 @@
+Incorrect ``threshold`` in ``np.set_printoptions`` raises ``TypeError`` or ``ValueError``
+-----------------------------------------------------------------------------------------
+Previously, an incorrect ``threshold`` raises ``ValueError``, it now raises ``TypeError``
```

Comment:
```suggestion
Previously an incorrect ``threshold`` raised ``ValueError``; it now raises ``TypeError``
```
Actually, now I am confused about the right grammar. Release notes get a final pass in any case though, so will merge without the fixup in a bit either way.

Reply:
you are right, it should be `raised`